File: codeabbey/013_sum_of_digits.rb
#!/usr/bin/env ruby
# encoding: utf-8

def sum_digits(num)
  def rec_sum(accumulator, num)
    return accumulator + num if num < 10
    accumulator += num % 10
    rec_sum accumulator, num / 10
  end
  rec_sum 0, num
end

x = ARGF.file.read.split "\n"
x.shift
x.map! do |e|
  e = e.split ' '
  e[0].to_i * e[1].to_i + e[2].to_i
end
puts x.map { |e| sum_digits(e) }.join ' '

File: codeabbey/013-sum-of-digits.js
#!/usr/bin/env node

var input, output;

input = require('fs')
        .readFileSync('./input.txt', 'utf8')
        .split('\n')
        .slice(1, -1)
        .map(function(nums) {
          return nums.split(' ')
          .map(function(num) {
            return parseInt(num, 10);
          });
        });

output = input.map(function(nums) {
  return (nums[0] * nums[1] + nums[2])
    .toString()
    .split('')
    .reduce(function(acc, e) {
      return acc + parseInt(e, 10);
    }, 0);
}).join(' ');

console.log(output);
